src/main/java/org/spongepowered/common/data/processor/data/entity/PassengerDataProcessor.java

```diff
@@ -0,0 +1,207 @@
+    public Optional<PassengerData> fill(DataContainer container, PassengerData passengerData) {
+        passengerData.set(Keys.BASE_VEHICLE, getData(container, Keys.BASE_VEHICLE));
+        passengerData.set(Keys.VEHICLE, getData(container, Keys.VEHICLE));
+        return Optional.of(passengerData);
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, PassengerData manipulator) {
+        if (dataHolder instanceof net.minecraft.entity.Entity) {
+            DataTransactionBuilder builder = DataTransactionBuilder.builder();
+            final Entity prevVehicle = (Entity) ((net.minecraft.entity.Entity) dataHolder).riddenByEntity;
+
+            net.minecraft.entity.Entity currentEntity = ((net.minecraft.entity.Entity) dataHolder).ridingEntity;
+
+            while (currentEntity != null && currentEntity.isRiding())
+            {
+                currentEntity = currentEntity.ridingEntity;
+            }
+
+            final Entity prevBaseVehicle = (Entity) currentEntity;
+            final net.minecraft.entity.Entity entity = (net.minecraft.entity.Entity) dataHolder;
+            final Entity newBaseVehicle = manipulator.baseVehicle().get();
+            final Entity newVehicle = manipulator.vehicle().get();
+
+            try {
+                builder.replace(
+                        new ImmutableSpongeValue<Entity>(Keys.VEHICLE, prevVehicle),
+                        new ImmutableSpongeValue<Entity>(Keys.BASE_VEHICLE, prevBaseVehicle));
+                entity.ridingEntity = (net.minecraft.entity.Entity) newVehicle;
+                builder.success(
+                        new ImmutableSpongeValue<Entity>(Keys.BASE_VEHICLE, newBaseVehicle),
+                        new ImmutableSpongeValue<Entity>(Keys.VEHICLE, newVehicle))
+                        .result(DataTransactionResult.Type.SUCCESS);
+                return builder.build();
+            } catch (Exception e) {
+                entity.riddenByEntity = (net.minecraft.entity.Entity) prevVehicle;
+                builder.reject(
+                        new ImmutableSpongeValue<Entity>(Keys.VEHICLE, newVehicle),
+                        new ImmutableSpongeValue<Entity>(Keys.BASE_VEHICLE, newBaseVehicle))
+                        .result(DataTransactionResult.Type.ERROR);
+                return builder.build();
+            }
+        }
+        return DataTransactionBuilder.failResult(manipulator.getValues());
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, PassengerData manipulator, MergeFunction function) {
+        return null;
+    }
+
+    @Override
+    public Optional<ImmutablePassengerData> with(Key<? extends BaseValue<?>> key, Object value, ImmutablePassengerData immutable) {
+        return Optional.absent();
+    }
+
+    @Override
+    public DataTransactionResult remove(DataHolder dataHolder) {
+        return DataTransactionBuilder.builder().result(DataTransactionResult.Type.FAILURE).build();
```

Comment:
Likewise here. Removing a `PassengerData` from an `Entity` should unmount the passenger.